disdat/path_cache.py

```diff
@@ -0,0 +1,134 @@
+#
+# Copyright 2015, 2016, 2017  Human Longevity, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+"""
+Disdat Luigi Interface file
+
+The path cache keeps a binding from luigi tasks to bundles.
+
+"""
+
+import collections
+
+from disdat import logger as _logger
+
+PathCacheEntry = collections.namedtuple('PathCacheEntry', 'instance bundle uuid path rerun')
+
+
+class PathCache(object):
+    """
+    PipeCache singelton
+
+    The path cache requires that everyone import "disdat.path_cache" and not "import path_cache".
+    Otherwise we will put entries in a class object whose name is '<class 'fs.DisdatFS'>' but search in an instance
+    of <class 'disdat.path_cache.PathCache'>. The driver uses the path_cache.PathCache class object, but when you run
+    from a pipe defined outside the project, then the system finds the class object with the package name appended.
+    And that class object doesn't have a path_cache with anything in it.
```

Comment:
not sure i understand the nuance, but seems very fragile

Reply:
Right.  Some clarifications.   First, no user should ever import this module directly, so that was a mistake in the comment, and I fixed it.   However, if the parts of Disdat that use it, namely pipe.py, pipe_base.py, and apply.py, do not import it in a fully-qualified, consistent manner, then Python can create multiple class objects (of the same class).